adaptive_memory_v4.0.py

```
@@ -0,0 +1,1982 @@
+            if not all_memories:
+                return False
+
+            if self.valves.use_embeddings_for_deduplication:
+                # Use embedding-based similarity
+                new_embedding = await self.embedding_manager.get_embedding(text)
+                if new_embedding is None:
+                    logger.warning("Could not generate embedding for duplicate check, falling back to text similarity")
+                    return await self._check_text_similarity(text, all_memories)
+                    
+                # Check similarity against existing memories
+                for i, memory in enumerate(all_memories):
```

Ruff (0.14.10)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)